neqo-transport/src/connection/mod.rs

```diff
@@ -2788,6 +2841,84 @@ impl Connection {
     pub fn stream_keep_alive(&mut self, stream_id: u64, keep: bool) -> Res<()> {
         self.streams.keep_alive(stream_id.into(), keep)
     }
+
+    fn max_dgram_size_configured(&self) -> Res<u64> {
+        let tph = self.tps.borrow();
+        if let Some(r) = &tph.remote {
+            Ok(r.get_integer(tparams::MAX_DATAGRAM_FRAME_SIZE))
+        } else if let Some(r) = &tph.remote_0rtt {
+            Ok(r.get_integer(tparams::MAX_DATAGRAM_FRAME_SIZE))
+        } else {
+            Err(Error::NotAvailable)
+        }
+    }
+
+    #[cfg(test)]
+    pub fn set_max_dgram_size(&mut self, v: u64) {
+        self.tps
+            .borrow_mut()
+            .remote
+            .as_mut()
+            .unwrap()
+            .set_integer(tparams::MAX_DATAGRAM_FRAME_SIZE, v);
+    }
+
+    /// Returns the current max size of a datagram that can fit into a packet.
+    /// # Error
+    /// The function returns `NotAvailable` if datagrams are not enabled.
```

Comment:
The problem with this function is that it is a prediction that is based on things that might change.  I have no problem with providing this function, but you should make it clear that this is an *estimate* of available capacity.  That goes with my comments about how DATAGRAM is prioritized for writing.

In practice, that means that sending DATAGRAM frames close to this limit is unwise.  We should skip any frames that don't fit.  We should avoid checking size in the send_datagram() function, absent a simple, hard maximum of 2^16-20 or thereabouts.

That means considering the "we dropped your datagram" event I already mentioned.  (That goes with a "your datagram was lost" event.)